stock_move_location/wizard/stock_move_location.py

```diff
@@ -0,0 +1,178 @@
+
+    def _get_collected_inventory_name(self):
+        sequence = self.env['ir.sequence'].next_by_code(
+            'stock.inventory.move') or '/'
+        res = "{sequence}:{location_from}:{location_to}".format(
+            sequence=sequence,
+            location_from=self.origin_location_id.display_name,
+            location_to=self.destination_location_id.display_name,
+        )
+        return res or "/"
+
+    def _get_collected_inventory_values(self):
+        return {
+            "name": self._get_collected_inventory_name(),
+            "location_id": self.origin_location_id.id,
+            "inventory_type": "move",
+            "destination_location_id": self.destination_location_id.id,
+            "filter": "partial",
+        }
+
+    def _get_inventory_action(self, inventory_id):
+        action = self.env.ref("stock.action_inventory_form").read()[0]
+        form_view = self.env.ref("stock.view_inventory_form").id
+        action.update({
+            "view_mode": "form",
+            "views": [(form_view, "form")],
+            "res_id": inventory_id,
+        })
+        return action
+
+    def _get_stock_move_location_lines_values(self):
+        quants_obj = self.env["stock.quant"]
+        product_obj = self.env['product.product']
+
+        locations = self.env['stock.location'].search([
+            ('id', 'child_of', [self.origin_location_id.id])
+        ])
+        company = self.env['res.company']._company_default_get(
+            'stock.inventory')
+        domain = [
+            ('location_id', 'in', locations.ids),
+            ('company_id', '=', company.id),
+        ]
+        grouped_quants_data = quants_obj.read_group(
+            domain=domain,
+            fields=['product_id', 'lot_id'],
+            groupby=['product_id', 'lot_id'],
+            lazy=False,
+        )
+
+        product_data = []
+        for group in grouped_quants_data:
+            quants = quants_obj.search(group.get("__domain"))
+            product = product_obj.browse(group.get("product_id")[0])
+            if group.get("lot_id"):
+                lot = group.get("lot_id")[0]
+            else:
+                lot = False
+            product_data.append(
+                {
+                    'product_id': product.id,
+                    'move_quantity': sum(quants.mapped("quantity")),
+                    'max_quantity': sum(quants.mapped("quantity")),
+                    'origin_location_id': self.origin_location_id.id,
+                    'destination_location_id': self.destination_location_id.id,
+                    'lot_id': lot,
+                    'product_uom_id': product.uom_id.id,
+                    'move_location_wizard_id': self.id,
+                }
+            )
+        return product_data
+
+    def add_lines(self):
+        self.ensure_one()
+        if not self.stock_move_locaiton_line_ids:
+            for line_val in self._get_stock_move_location_lines_values():
```

Comment:
how many records do we expect to create potentially? This can be reaaaaally heavy w/ all the writes and the computations after hand. We could do a multiple insert via SQL then trigger recompute on newly created records.

Reply:
@mpanarin any answer to this?

Reply:
To be honest - I don't know about the number of lines. That depends on the workflow of the end user. 

I minimized the amount of creates. So here - only lines of the wizards are created and nothing more. As well as i switched from inventories to pickings, which are less heavy. 

Switching to sql inserts is not worth it in my opinion anyway :\

Reply:
@simahawk I think this operation is not an usual one. If your stock locations are fine tuned (just some product or just one in a same location), the solution is quite acceptable